include/seqan3/alignment/pairwise/align_result.hpp

```diff
@@ -34,210 +34,157 @@
 
 /*!\file
  * \brief Provides seqan3::align_result.
+ * \author Jörg Winkler <j.winkler AT fu-berlin.de>
  * \author Rene Rahn <rene.rahn AT fu-berlin.de>
  */
 
 #pragma once
 
-#include <functional>
-#include <optional>
-#include <tuple>
-
-#include <seqan3/core/metafunction/template_inspection.hpp>
-
 namespace seqan3
 {
 
-/*!\brief Keys for different alignment results.
- * \ingroup pairwise
- */
-enum struct align_result_key : uint8_t
+namespace detail
 {
-    id,     // cannot be set by the config
-    score,  // report the score
-    end,    // report the end position and score
-    begin,  // report the begin and end position and score
-    trace   // report the full trace and begin and end and score
-};
 
-/*!\brief Stores the alignment results and offers a tuple-like interface.
- * \extends std::tuple
- * \ingroup pairwise
- * \tparam output_type_list_t seqan3::type_list over the contained results.
+/*!
+ * \brief A struct that contains the actual alignment result data.
+ * \tparam id_t         The type for the alignment identifier.
+ * \tparam score_t      The type for the resulting score.
+ * \tparam coordinate_t The type for both the begin and end coordinates, can be omitted.
+ * \tparam trace_t      The type for the alignment trace, can be omitted.
  */
-template <typename output_type_list_t>
-struct align_result //!\cond
-    : public detail::transfer_template_args_onto_t<output_type_list_t, std::tuple>
-    //!\endcond
+template <typename id_t, typename score_t, typename coordinate_t = void *, typename trace_t = void *>
+struct align_result_value_type
 {
+    //! \brief The alignment identifier.
+    id_t id{};
+    //! \brief The alignment score.
+    score_t score{};
+    //! \brief The end coordinate of the alignment.
+    coordinate_t end_coordinate{};
+    //! \brief The begin coordinate of the alignment.
+    coordinate_t begin_coordinate{};
```

Comment:
should be a different type name, as it would be possible to compute the end point without the begin point. But if they have the same type, I can not check with static_assert anymore, if it was computed or not.